src/common/config.cc

```diff
@@ -1227,6 +1227,46 @@ void md_config_t::diff(
   }
 }
 
+void md_config_t::diff_setting(const md_config_t *other, 
+            map<string, pair<string, string>> *diff, 
```

Comment:
as we discussed, in your case, at most only one item will be returned. so either we can consolidate these two diff() methods, or we don't need to use a `map` here.